mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -56,10 +67,22 @@ protected void onCreate(Bundle savedInstanceState) {
         generatedTV = findViewById(R.id.sample_text);
         promptTV = findViewById(R.id.user_text);
 
+        Consumer<String> tokenListener = this;
+
+        //enable scrolling and resizing of text boxes
+        generatedTV.setMovementMethod(new ScrollingMovementMethod());
+        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
+
         // adding on click listener for send message button.
         sendMsgIB.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                if (model == null) {
```

Comment:
should we just disable the button until the model is ready?

Reply:
Button is enabled to give error message that model is still downloading to prevent app from crashing. I made a download progress meter on the Phi-3 Vision app that replaced this functionality which can be brought over but would need to update UX to include this.

Reply:
`model` and `tokenizer` are possibly set by another thread, i.e., when the download complete callback initializes them.

we rely on both of them to be initialized here, but it is possible this check is hit (`if (model != null)`) between the assignment to `model` and the assignment to `tokenizer`.

disabling the button until they are ready is one simple way to avoid thread-safety issues. alternatively, maybe making those variables `volatile` and checking both `model` and `tokenizer` here would also work.

Reply:
You are right, I'll replace with tokenizer == null comparison to make sure we don't hit a race condition. This comparison should be atomic so I don't think we need to move allocator to UI thread.